SU-SWS/decanter-react
# ADAPT-2142: Dismiss Button as its own component and Alert tweaks

# READY FOR REVIEW

# Summary
- Add Dismiss Button as its own component since we often need one in modals and other components.
- Use it in the Alert component

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Look at code
1. Look at preview build and see the Dismiss Button examples
2. Look at the Alert component and see that alignment of labels, content, dismiss button should be improved. Also spacing has been tweaked - please check the different examples at different breakpoints.

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
